File: web/src/main/java/hu/bsstudio/robonaut/scores/AudienceScoreHandler.java
package hu.bsstudio.robonaut.scores;

import hu.bsstudio.robonaut.scores.audience.AudienceScoreService;
import hu.bsstudio.robonaut.scores.audience.model.AudienceScoredTeam;
import hu.bsstudio.robonaut.team.model.DetailedTeam;
import lombok.RequiredArgsConstructor;
import org.springframework.web.reactive.function.server.HandlerFunction;
import org.springframework.web.reactive.function.server.ServerRequest;
import org.springframework.web.reactive.function.server.ServerResponse;
import reactor.core.publisher.Mono;

@RequiredArgsConstructor
public class AudienceScoreHandler implements HandlerFunction<ServerResponse> {

    private final AudienceScoreService service;

    @Override
    public Mono<ServerResponse> handle(final ServerRequest request) {
        final var detailedTeam = request.bodyToFlux(AudienceScoredTeam.class)
            .flatMap(service::updateAudienceScore);
        return ServerResponse.ok().body(detailedTeam, DetailedTeam.class);
    }
}

File: web/src/main/kotlin/hu/bsstudio/robonaut/scores/AudienceScoreHandler.kt
package hu.bsstudio.robonaut.scores

import hu.bsstudio.robonaut.scores.audience.AudienceScoreService
import hu.bsstudio.robonaut.scores.audience.model.AudienceScoredTeam
import org.springframework.web.reactive.function.server.*
import reactor.core.publisher.Mono

class AudienceScoreHandler(private val service: AudienceScoreService) : HandlerFunction<ServerResponse> {

    override fun handle(request: ServerRequest): Mono<ServerResponse> {
        return request.bodyToFlux<AudienceScoredTeam>()
            .flatMap(service::updateAudienceScore)
            .let { detailedTeam -> ServerResponse.ok().body(detailedTeam) }
    }
}
